components/hab/src/command/origin/create.rs

```diff
@@ -0,0 +1,32 @@
+use crate::{api_client::{self,
+                         Client},
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+use hyper::status::StatusCode;
+
+pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Creating, format!("origin {}.", origin))?;
+
+    match api_client.create_origin(origin, token) {
+        Ok(_) => {
+            ui.status(Status::Created, format!("origin {}.", origin))?;
+            Ok(())
+        }
+        Err(api_client::Error::APIError(StatusCode::Conflict, _msg)) => {
+            ui.status(Status::Skipping,
+                      format!("creation of origin {}. Origin already exists!", origin))?;
+            Ok(())
```

Comment:
I think it makes more sense to make this an error from the standpoint of automation. Outputting a different exit code would also be useful. It's easier to intentionally ignore a particular kind of error than find one based on output our code considers "ok". If people really desire the behavior that trying to create an existing origin is not an error, we can always add an additional flag that lets people opt into that behavior.

Reply:
This is consistent with commands like origin key upload, and the automation we're adding this for will be much easier if we just follow that pattern.  We can do more complicated things later..

Reply:
100% agree that consistency is the best thing. Just wanted to point out that in general it's *way* harder to make something that succeeds fail later than the other way around. Since we're adding new functionality here, let's be careful we avoid YAGNI, but also don't make choices that paint us into corners.